### ISSUE TITLE
Callback prop for dismissible notifications

### ISSUE THREAD

**Issue by mujtabahasni** [telus/tds-core#1056]
## Description
I want my app to know when the customer has dismissed a dismissible notification generated within the app.  

When the customer dismisses a notification, it is removed from the DOM but there is no way for the app to know when this happens.   

## Recommendation
Provide a callback prop to be used with dismissible notifications, such as `onDismiss`.  When the customer dismisses a notification, the callback prop will be called.

## Workarounds
My current work around is to create a non-dismissible notification use inside, add `<StandAloneIcon symbols="times" onClick={myCallback}/>`

## Meta

- TDS component: @tds/core-notification
- Willing to develop solution: Yes
- Has workaround: Yes
- High impact: No
- Designer/Developer Pair: Kamal Shatila / Mujtaba Hasni

**Comment by theetrain** [comment]
This is a great suggestion. We welcome a pull request; I recommend adding an `onDismiss` prop, `onCloseClick` prop, or both!

```jsx
Notification.propTypes = {
  // A callback that invokes when the user clicks on the close button, requires `dimissible={true}
  onCloseClick: PropTypes.func,
  // A callback that invokes when the Notification component fully dismisses and unmounts
  onDismiss: PropTypes.func
}
```

**Comment by agorovyi** [comment]
This has been released in [@tds/core-notification@1.3.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-notification%401.3.0)
